drivers/memc/memc_stm32_xspi_psram.c

```diff
@@ -0,0 +1,363 @@
+/*
+ * Copyright (c) 2025 STMicroelectronics
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT st_stm32_xspi_psram
+
+#include <errno.h>
+#include <soc.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/clock_control.h>
+#include <zephyr/drivers/clock_control/stm32_clock_control.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(memc_stm32_xspi_psram, CONFIG_MEMC_LOG_LEVEL);
+
+#define STM32_XSPI_NODE DT_INST_PARENT(0)
+
+/* Macro to check if any xspi device has a domain clock or more */
+#define STM32_XSPI_PSRAM_DOMAIN_CLOCK_INST_SUPPORT(inst)	\
+	DT_CLOCKS_HAS_IDX(DT_INST_PARENT(inst), 1) ||
+#define STM32_XSPI_PSRAM_INST_DEV_DOMAIN_CLOCK_SUPPORT		\
+	(DT_INST_FOREACH_STATUS_OKAY(STM32_XSPI_PSRAM_DOMAIN_CLOCK_INST_SUPPORT) 0)
+
+/* This symbol takes the value 1 if device instance has a domain clock in its dts */
+#if STM32_XSPI_PSRAM_INST_DEV_DOMAIN_CLOCK_SUPPORT
+#define STM32_XSPI_PSRAM_DOMAIN_CLOCK_SUPPORT 1
+#else
+#define STM32_XSPI_PSRAM_DOMAIN_CLOCK_SUPPORT 0
+#endif
```

Comment:
I'm not sure this whole mechanism has a real interest in this specific case as it will save only a few bytes which may be unsignificant on firmwares making use of external RAM